sdk/storage/azure-storage-common/pom.xml

```diff
@@ -93,6 +93,12 @@
       <version>1.15.2</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
```

Comment:
CI is failing because `mockito-inline:4.11.0` is really old and leverages older versions of code that performs deep reflection that doesn't work on newer versions of Java without including overrides of those dependency. You'd want to copy what's being done here:

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-http-vertx/pom.xml#L140

But in this case, from reviewing the PR, you don't even need Mockito for mocking as all the types you're using can be manually mocked or are simple interfaces. See comments below.